native/dgeqrf.go

```diff
@@ -25,14 +25,14 @@ func (impl Implementation) Dgeqrf(m, n int, a []float64, lda int, tau, work []fl
 	nb := impl.Ilaenv(1, "DGEQRF", " ", m, n, -1, -1)
 	lworkopt := n * max(nb, 1)
 	lworkopt = max(n, lworkopt)
+	if len(work) < max(1, lwork) {
```

Comment:
Done.

I'm going to file an issue to consider moving checkMatrix calls higher universally as well. The decision may be not to, but I think it's something that should be discussed: #228.